lib/msf/core/handler.rb

```diff
@@ -224,16 +224,26 @@ def create_session(conn, opts={})
       # Pass along any associated payload uuid if specified
       if opts[:payload_uuid]
         s.payload_uuid = opts[:payload_uuid]
-        payload_info = {
-            uuid: s.payload_uuid.puid_hex,
-            workspace: framework.db.workspace
-        }
-        if s.payload_uuid.respond_to?(:puid_hex) && (uuid_info = framework.db.payloads(payload_info).first)
-          s.payload_uuid.registered = true
-          s.payload_uuid.name = uuid_info['name']
-          s.payload_uuid.timestamp = uuid_info['timestamp']
-        else
-          s.payload_uuid.registered = false
+        begin
+          payload_info = {
+              uuid: s.payload_uuid.puid_hex,
+              workspace: framework.db.workspace
+          }
+          if s.payload_uuid.respond_to?(:puid_hex) && (uuid_info = framework.db.payloads(payload_info).first)
```

Comment:
I'm all for robust error handling code, however, you might be able to clean this up a little by checking `framework.db.active`.
```
if s.payload_uuid.respond_to?(:puid_hex) && framework.db.active && (uuid_info = framework.db.payloads(payload_info).first)
```